ldms/src/ldmsd/ldmsd.c

```diff
@@ -2173,25 +2194,7 @@ int main(int argc, char *argv[])
 		exit(1);
 	}
 
-	if (!foreground) {
-		/* Create pidfile for daemon that usually goes away on exit. */
-		/* user arg, then env, then default to get pidfile name */
-		if (!pidfile) {
-			char *pidpath = getenv("LDMSD_PIDFILE");
-			if (!pidpath) {
-				pidfile = malloc(strlen(LDMSD_PIDFILE_FMT)
-						+ strlen(basename(argv[0]) + 1));
-				if (pidfile)
-					sprintf(pidfile, LDMSD_PIDFILE_FMT, basename(argv[0]));
-			} else {
-				pidfile = strdup(pidpath);
-			}
-			if (!pidfile) {
-				ovis_log(NULL, OVIS_LERROR, "Out of memory\n");
-				av_free(auth_opt);
-				exit(1);
-			}
-		}
+	if (pidfile) {
 		if( !access( pidfile, F_OK ) ) {
 			ovis_log(NULL, OVIS_LERROR, "Existing pid file named '%s': %s\n",
```

Comment:
@nichamon, @valleydlr do we need this message?

Reply:
It seems this message could be helpful in identifying misconfiguration of the `-r` in LDMSD. For example, if two LDMSDs are configured to create a PID file at `/home/my/var/pid/ldmsd.pid` on different nodes that share the same home directory, they will overwrite each other's PID files.This scenario is more likely to occur in a test environment. Ultimately, I am neutral on whether to keep or remove the message.

Reply:
@nichamon, I think this is reasonable.